crates/primitives/src/utilities.rs

```diff
@@ -31,6 +33,56 @@ pub fn create2_address(caller: B160, code_hash: B256, salt: U256) -> B160 {
     B160(hasher.finalize().as_slice()[12..].try_into().unwrap())
 }
 
+/// Calculates the `excess_blob_gas` from the parent header's `blob_gas_used` and `excess_blob_gas`.
+///
+/// See also [the EIP-4844 helpers](https://eips.ethereum.org/EIPS/eip-4844#helpers).
+#[inline]
+pub fn calc_excess_blob_gas(parent_excess_blob_gas: u64, parent_blob_gas_used: u64) -> u64 {
+    (parent_excess_blob_gas + parent_blob_gas_used).saturating_sub(TARGET_BLOB_GAS_PER_BLOCK)
+}
+
+/// Calculates the blobfee from the header's excess blob gas field.
+///
+/// See also [the EIP-4844 helpers](https://eips.ethereum.org/EIPS/eip-4844#helpers).
+#[inline]
+pub fn calc_blob_fee(excess_blob_gas: u64) -> u64 {
+    fake_exponential(
+        MIN_BLOB_GASPRICE,
+        excess_blob_gas,
+        BLOB_GASPRICE_UPDATE_FRACTION,
+    )
+}
+
+/// Approximates `factor * e ** (numerator / denominator)` using Taylor expansion.
+///
+/// This is used to calculate the blob price.
+///
+/// See also [the EIP-4844 helpers](https://eips.ethereum.org/EIPS/eip-4844#helpers).
+#[inline]
+pub fn fake_exponential(factor: u64, numerator: u64, denominator: u64) -> u64 {
+    assert!(denominator > 0, "attempt to divide by zero");
```

Comment:
would panic here if dominator is == 0

Reply:
Should we return 0/None if `denominator == 0`? It would be divide-by-zero

Reply:
I would just `panic!` as it is undefined what to expect as result, maybe it is `u64::MAX` maybe `0`

Reply:
a panic is fine here imo, for blob fee, the denominator is a constant != 0